llama-index-packs/llama-index-packs-secgpt/llama_index/packs/secgpt/planner.py

```diff
@@ -0,0 +1,86 @@
+"""
+The hub planner accepts inputs including queries, tool information, and chat history to create a plan that outlines the necessary tools and data. It can be tailored with various prompt templates and an output parser to specifically customize the content and format of the generated plan.
+"""
+
+from llama_index.core.query_pipeline import QueryPipeline
+from llama_index.core.llms.llm import LLM
+from llama_index.core.output_parsers import LangchainOutputParser
+from langchain_core.output_parsers import JsonOutputParser
+
+from llama_index.core.llms import ChatMessage, MessageRole
+from llama_index.core import ChatPromptTemplate
+
+
+class HubPlanner:
```

Comment:
Since there is a prompt here, I think we should subclass `PromptMixin`:

https://github.com/run-llama/llama_index/blob/main/llama-index-core/llama_index/core/prompts/mixin.py

Reply:
I have refined the prompt definition with a subclass of PromptMixin.